### ISSUE TITLE
Error : Functions found when exporting methods from the namespace 'Matrix' which are not S4 generic: 'as.array', 'as.matrix', 'mean'

### ISSUE THREAD

**Issue by dorianps** [stnava/ANTsR#165]
Sorry to flood with issues, but since yesterday I haven't been able to upgrade ANTsR in our cluster. So far, I haven't been able to fix it, after removing and reinstalling ITKR/ANTsRCore. Before I try other things, does this error look familiar to anyone.

R is v3.2.5
```
sessionInfo()
R version 3.2.5 (2016-04-14)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS release 6.8 (Final)

locale:
 [1] LC_CTYPE=en_US.iso885915       LC_NUMERIC=C
 [3] LC_TIME=en_US.iso885915        LC_COLLATE=en_US.iso885915
 [5] LC_MONETARY=en_US.iso885915    LC_MESSAGES=en_US.iso885915
 [7] LC_PAPER=en_US.iso885915       LC_NAME=C
 [9] LC_ADDRESS=C                   LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.iso885915 LC_IDENTIFICATION=C

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base

loaded via a namespace (and not attached):
[1] tools_3.2.5
>
```

```
Warning: namespace 'ANTsR' is not available and has been replaced
by .GlobalEnv when processing object 'plot.index'
Warning: namespace 'ANTsR' is not available and has been replaced
by .GlobalEnv when processing object 'plot.index'
Warning: namespace 'ANTsR' is not available and has been replaced
by .GlobalEnv when processing object 'plot.index'
Warning: namespace 'ANTsR' is not available and has been replaced
by .GlobalEnv when processing object 'plot.index'
Warning: namespace 'ANTsR' is not available and has been replaced
by .GlobalEnv when processing object 'plot.index'
Warning: namespace 'ANTsR' is not available and has been replaced
by .GlobalEnv when processing object 'plot.index'
Warning: namespace 'ANTsR' is not available and has been replaced
by .GlobalEnv when processing object 'plot.index'
Warning: namespace 'ANTsR' is not available and has been replaced
by .GlobalEnv when processing object 'plot.index'
Warning: namespace 'ANTsR' is not available and has been replaced
by .GlobalEnv when processing object 'plot.index'
Warning: namespace 'ANTsR' is not available and has been replaced
by .GlobalEnv when processing object 'plot.index'
Warning: namespace 'ANTsR' is not available and has been replaced
by .GlobalEnv when processing object 'plot.index'
Warning: namespace 'ANTsR' is not available and has been replaced
by .GlobalEnv when processing object 'plot.index'
Warning: namespace 'ANTsR' is not available and has been replaced
by .GlobalEnv when processing object 'plot.index'
Warning: namespace 'ANTsR' is not available and has been replaced
by .GlobalEnv when processing object 'plot.index'
Warning: namespace 'ANTsR' is not available and has been replaced
by .GlobalEnv when processing object 'plot.index'
Warning: namespace 'ANTsR' is not available and has been replaced
by .GlobalEnv when processing object 'plot.index'
Error : Functions found when exporting methods from the namespace 'Matrix' which are not S4 generic: 'as.array', 'as.matrix', 'mean'
ERROR: lazy loading failed for package 'ANTsRCore'
* removing '/home/dpustina/R/x86_64-unknown-linux-gnu-library/3.1/ANTsRCore'

```

**Comment by stnava** [comment]
is there a conflict of R versions?  you seem to have

R version 3.2.5 (2016-04-14)


as well as

/home/dpustina/R/x86_64-unknown-linux-gnu-library/3.1/




brian

**Comment by dorianps** [comment]
Yeah, that might have been the source of a number of error I used to receive. This is long but I am putting it there in case someone has similar issues.

Sometimes in the past I must have switched to R 3.2, then installed devtools. The library path was kept the same and pointed at folders 3.1 to avoid installing all packages each time I switched R versions. The problem with 3.2 was that Rstudio had a bug that didn't display the documentation properly. So I switched back to 3.1 and worked mainly there, everything worked fine with ANTsR. Last week I wanted to try devtools installation for LESYMAP, and it failed in R3.1 because `trimws` call (a 3.2 function) was missing. So I switched to 3.2, keeping all the time the path pointed at the same 3.1 libraries.

Long story short, now I resolved it by switching back to R v3.1, uninstalling 'Matrix', resintalling 'Matrix' again (with their special repo to allow installation `repos="http://R-Forge.R-project.org"`), then installing in series ANTsRCore and ANTsR. I also uninstalled devtools and reinstalled them from 3.1, and now seems to be working (it's installing ITKR).
